### PR TITLE
[RFC] Add fast/accurate flag to scan config

### PR DESCRIPTION
**Pull request by jaimejiyepark**

### CODE REVIEW

**Review by awushensky — APPROVED** [review]

File: CardScan/CardScan/Classes/ScanConfiguration.swift [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import Foundation
 
+@objc public enum ScanPerformance: Int {
+    case fastScan
```
Comment (kingst) [inline CardScan/CardScan/Classes/ScanConfiguration.swift:4]: Nit: I'd call these `fast` and `accurate` -- drop the Scan

File: CardScan/CardScan/Classes/ScanConfiguration.swift [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+    case accurateScan
+}
+
 @objc public class ScanConfiguration: NSObject {
     @objc public var runOnOldDevices = false
     @objc public var setPreviouslyDeniedDevicesAsIncompatible = false
+    @objc public static var scanPerformancePriority: ScanPerformance = .fastScan
```
Comment (kingst) [inline CardScan/CardScan/Classes/ScanConfiguration.swift:11]: I wouldn't make this a static configuration variable -- it should be a member variable on `SimpleScanViewController` that defaults to `fast` and people can pick `accurate` on a per viewcontroller basis. For example, if someone is using both scan and verify they might want different configurations for the two.

**Review by kingst — COMMENTED** [review]
I wouldn't set this as a static variable. Also, is there a simple unit test that might make sense for this? Any state machine updates make me nervous.

**Comment by jaimejiyepark** [comment]
@kingst I made changes in that instead of changing `OcrMainLoopStateMachine` , `SimpleScan` will use the new `OcrDurationMainLoopStateMachine` (which is the modified custom state machine we sent to verif-y and gsa)  when set to `.accurate`.

File: CardScan/CardScan/Classes/CreditCardOcr/MainLoopStateMachine.swift [hunk 3]
```diff
@@ -67,3 +66,51 @@ public protocol MainLoopStateMachine {
         return OcrMainLoopStateMachine()
     }
 }
+
+@objc public class OcrDurationMainLoopStateMachine: NSObject, MainLoopStateMachine {
```
Comment (kingst) [inline CardScan/CardScan/Classes/CreditCardOcr/MainLoopStateMachine.swift:70]: Nit: Maybe call this `OcrAccurateMainLoopStateMachine` or something like that

File: CardScan/CardScan/Classes/ScanConfiguration.swift [hunk 4]
```diff
@@ -1,5 +1,10 @@
 import Foundation
 
+@objc public enum ScanPerformancePriority: Int {
```
Comment (kingst) [inline CardScan/CardScan/Classes/ScanConfiguration.swift:3]: Nit: Maybe call this `ScanPerformance`

**Review by kingst — APPROVED** [review]
A few optional comments, but ship it.

Can you add another button to the example app to test out `accurate` mode?

After you address these ship it

**Pull request merged**